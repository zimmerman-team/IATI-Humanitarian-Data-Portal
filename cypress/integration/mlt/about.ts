/// <reference types="cypress" />

import autoRecord from 'cypress-autorecord';

const aboutTexts = [
  'The Grand Bargain transparency commitment',
  'What is the Grand Bargain?',
  'What is the Grand Bargain transparency commitment?',
  'What is IATI and why is it part of the transparency commitment?',
];

describe('About', function() {
  // autoRecord();

  it('should load the page', function() {
    cy.visit('/about');
  });

  /*it('image snapshot', function() {
    cy.document().toMatchImageSnapshot();
  });*/

  /*it('html snapshot', function() {
    cy.findAllByTestId('paragraph')
      .should('exist')
      .toMatchImageSnapshot();
  });*/

  it('should show the correct title', function() {
    aboutTexts.map(text => cy.queryByText(text).should('exist'));
  });

  it('should show background decoration', function() {
    cy.findByTestId('DecoAboutTopLeft').should('exist');
    cy.findByTestId('DecoAboutRight').should('exist');
    cy.findByTestId('DecoAboutMidRight').should('exist');
    cy.findByTestId('DecoAboutMidLeft').should('exist');
    cy.findByTestId('DecoAboutBottomRight').should('exist');
  });

  it('should show footer and the correct content', function() {
    cy.findByTestId('Footer').should('exist');
    cy.findByTestId('DI link').should('exist');
    cy.findByTestId('GOV link').should('exist');
    cy.findByTestId('GB link').should('exist');
  });
});
