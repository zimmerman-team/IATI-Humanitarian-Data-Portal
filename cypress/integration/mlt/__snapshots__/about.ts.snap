exports[`About page > should show the correct title #0`] = `
<h3 class="MuiTypography-root MuiTypography-h3 MuiTypography-colorTextPrimary">
  The Grand Bargain transparency commitment
</h3>
`;

exports[`About page > should show the correct title #1`] = `
<h6 class="MuiTypography-root MuiTypography-h6">What is the Grand Bargain?</h6>
`;

exports[`About page > should show the correct title #2`] = `
<h6 class="MuiTypography-root MuiTypography-subtitle1">
  What is the Grand Bargain transparency commitment?
</h6>
`;

exports[`About page > should show the correct title #3`] = `
<h6 class="MuiTypography-root MuiTypography-h6">
  What is IATI and why is it part of the transparency commitment?
</h6>
`;
